### ISSUE TITLE
Trouble with sending the printk messages to the dev console.  

### ISSUE THREAD

**Issue by mightylastingcode** [ALIBERA/linux_raspberrypi_book#3]
Hi Alberto,

I am unable to make the printk messages displayed in the terminal windows..    I wonder how you did it.

But, I found out that if I boot up the Pi OS in the text mode. it works.   The terminal in the text mode is /dev/console.   I confirmed it by using the tty command.   

After the bootup,

Type tty at the command prompt to verify that it is indeed /dev/console.

$tty /dev/tty1

Because I set console=tty1 in /boot/cmdline.txt file, I get /dev/tty1.  Now all the printk messages will appear immediately without dmesg.

Is this how you did it?   All my terminal window in GUI is not /dev/tty1.     The instructions in chapter 1 are not clear to me.

Please clarify if you can.  Thank you.

Michael

**Comment by ALIBERA** [comment]
Hi Michael,

Are you using a serial to USB converter connected to the UART pins?

Amazon.com : raspberry pi serial cable<https://www.amazon.com/raspberry-pi-serial-cable/s?k=raspberry+pi+serial+cable>

***@***.***

Best Regards,
Alberto


Alberto Liberal De Los Rios
Senior Technology Field Application Engineer Emb. & Connectivity
T: +34 917 61 11 71|M: +34 699 98 43 26
E: ***@***.***
Arrow Electronics | arrow.com
De: ML ***@***.***>
Enviado el: sábado, 4 de marzo de 2023 3:36
Para: ALIBERA/linux_raspberrypi_book ***@***.***>
CC: Subscribed ***@***.***>
Asunto: [External] [ALIBERA/linux_raspberrypi_book] Trouble with sending the printk messages to the dev console. (Issue #3)


CAUTION: This email originated from outside of the organization. This message might not be safe, use caution in opening it. If in doubt, do not open the attachment nor links in the message.


Hi Alberto,

I am unable to make the printk messages displayed in the terminal windows.. I wonder how you did it.

But, I found out that if I boot up the Pi OS in the text mode. it works. The terminal in the text mode is /dev/console. I confirmed it by using the tty command.

After the bootup,

Type tty at the command prompt to verify that it is indeed /dev/console.

$tty /dev/tty1

Because I set console=tty1 in /boot/cmdline.txt file, I get /dev/tty1. Now all the printk messages will appear immediately without dmesg.

Is this how you did it? All my terminal window in GUI is not /dev/tty1. The instructions in chapter 1 are not clear to me.

Please clarify if you can. Thank you.

Michael

-

**Comment by mightylastingcode** [comment]
Thank you.    I saw the picture in the instruction pdf file where you connect an USB-to-UART cable to Pi.   I did not understand why you do that.

When I examined /boot/cmdlines.txt again, I realized that the console is also set to be the serial terminal.

console=serial0,115200 console=tty1 root=PARTUUID=8c74060c-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait loglevel=8

That means that the console can be either accessed by the serial0 port or the tty1 terminal.    Inside the GUI desktop, those are terminal emulators (/dev/pts/0).  They are not tty1.   They cannot show the printk messages!   I was stuck there for a while.

Now I know I can use Alt + Ctrl + F1 to switch from GUI to the tty1 console text terminal.   I will be able to see the prinkt messages 

Or,  I use a remote PC to login to PI via the serial 0 port.  In the serial terminal program (putty), I can see all the system messages (the printk messages) displayed.  

You are right.

Thank you
Michael
